src/dali/plugin/input/rest/binance_com.py

```diff
@@ -0,0 +1,1295 @@
+                    fiat_in_with_fee=str(transaction[_SOURCEAMOUNT]),
+                    fiat_fee=str(RP2Decimal(transaction[_TOTALFEE])),
+                    fiat_ticker=transaction[_FIATCURRENCY],
+                    notes=(f"Buy transaction for fiat payment orderNo - " f"{transaction[_ORDERNO]}"),
+                )
+            )
+
+        else:
+            trade: _Trade = self._to_trade(transaction[_SYMBOL], str(transaction[_AMOUNT]), str(transaction[_COST]))
+            if transaction[_SIDE] == _BUY:
+                out_asset = trade.quote_asset
+                in_asset = trade.base_asset
+                crypto_in = RP2Decimal(str(transaction[_AMOUNT]))
+                conversion_info = f"{trade.quote_info} -> {trade.base_info}"
+            elif transaction[_SIDE] == _SELL:
+                out_asset = trade.base_asset
+                in_asset = trade.quote_asset
+                crypto_in = RP2Decimal(str(transaction[_COST]))
+                conversion_info = f"{trade.base_info} -> {trade.quote_info}"
+            else:
+                raise Exception(f"Internal error: unrecognized transaction side: {transaction[_SIDE]}")
+
+            if transaction[_FEE][_CURRENCY] == in_asset:
+                crypto_fee = RP2Decimal(str(transaction[_FEE][_COST]))
+            else:
+                crypto_fee = ZERO
+
+                transaction_fee = transaction[_FEE][_COST]
+
+                # Users can use BNB to pay fees on Binance
+                if transaction[_FEE][_CURRENCY] != out_asset and float(transaction_fee) > 0:
+                    out_transaction_list.append(
+                        OutTransaction(
+                            plugin=self.__BINANCE_COM,
+                            unique_id=transaction[_ID],
+                            raw_data=json.dumps(transaction),
+                            timestamp=self._rp2timestamp_from_ms_epoch(transaction[_TIMESTAMP]),
+                            asset=transaction[_FEE][_CURRENCY],
+                            exchange=self.__BINANCE_COM,
+                            holder=self.account_holder,
+                            transaction_type=Keyword.FEE.value,
+                            spot_price=Keyword.UNKNOWN.value,
+                            crypto_out_no_fee="0",
+                            crypto_fee=str(transaction_fee),
+                            crypto_out_with_fee=str(transaction_fee),
+                            fiat_out_no_fee=None,
+                            fiat_fee=None,
+                            notes=(f"{notes + '; ' if notes else ''} Fee for conversion from " f"{conversion_info}"),
+                        )
+                    )
+
+            # Is this a plain buy or a conversion?
+            if trade.quote_asset in self.client.options[_LEGALMONEY]:  # Binance specific param
+                fiat_in_with_fee = RP2Decimal(str(transaction[_COST]))
+                fiat_fee = RP2Decimal(crypto_fee)

```

Comment:
My understanding is that fiat buys are treated like a regular buy/sell, just that the quote asset is fiat. So, the `crypto_in` or `crypto_out` is actually fiat in and fiat out. And then the fiat amounts should be the same since they are there to represent the fiat value of the asset. Is that not correct? Maybe it would be easier for you to look at the tests ([the inputs](https://github.com/macanudo527/dali-rp2/blob/450b22d15ce47acda82ebe68b84cb2a6b0baaa21/tests/test_plugin_binance_com.py#L261-L301) and [the assertions](https://github.com/macanudo527/dali-rp2/blob/450b22d15ce47acda82ebe68b84cb2a6b0baaa21/tests/test_plugin_binance_com.py#L409-L451)).

Honestly, I don't do any fiat buy/sells on Binance.com so it is very difficult for me to see this with real data. I thought I had it right though with the tests.

Reply:
I should mention `self.client.options[_LEGALMONEY]` is a list of all fiat that has ever been traded on Binance.com.

Reply:
Maybe add a comment that crypto and fiat are the same in this case.